lib/utils/utils.js

```diff
@@ -64,8 +67,49 @@ function setConsoleLogPrefix(prefix) {
     };
 }
 
+// This should be roughly similar to ARG_MAX, some systems is 131072 per argument 
+// but we're going to be very conservative and cap the total size to be safe.
+const CMD_SIZE_LIMIT = 8000;
+
+/**
+ * Only allows a specific number of bytes for variable names and values.  Any variable exceeding that will be stored in a file
+ * and the variable value will contain the temp file location.
+ * If this substitution occurs then an additional variable called FILE_PARAMS will contain the list of substituted variable names.
+ * @param {Object} vars 
+ * @param {Number} sizeLimit Size limit to uphold
+ * @returns Object of values from the provided one, modified as described.
+ */
+function limitVariableSizes(vars, sizeLimit = CMD_SIZE_LIMIT) {
+    // First sort variables by serialized (string representation) size so we get the smallest variables out first
+    const keys = Object.keys(vars).sort((a,b) => 
+        String(vars[a]).length - String(vars[b]).length
+    );
+    const sizeLimitedVars = {};
+    let totalSize = 0;
+    for (const key of keys) {
+        const newSize = totalSize + String(key).length + String(vars[key]).length;
+        if (newSize >= sizeLimit) {
+            // We have hit our limit and all variables must be stored in temp files now
+            const tmpFile = tmp.fileSync();
```

Comment:
As is, I assume the temp file will only be removed once the node process exits. But it should be removed after the current activation finishes. To achieve that, a temp file can be created inside the current activation folder - which is unique and will already be automatically removed after the processing.

Reply:
Oh!  That's right.  Good point, thank you.